server/plugin.go

```diff
@@ -875,13 +875,19 @@ func (p *Plugin) GetRemoteID() string {
 }
 
 func (p *Plugin) updateMetrics() {
-	stats, err := p.store.GetStats()
+	start := time.Now()
+	p.API.LogDebug("Updating metrics")
+	stats, err := p.store.GetStats(p.remoteID, PreferenceCategoryPlugin)
 	if err != nil {
 		p.API.LogWarn("failed to update computed metrics", "error", err)
+		return
```

Comment:
Because `GetStats()` returns nil in case of error, we should return here otherwise the calls below will panic